bevy_lint/src/sym.rs

```diff
@@ -0,0 +1,186 @@
+//! Pre-interned [`Symbol`]s available in `const` contexts.
+//!
+//! [`Symbol`]s are [interned strings](https://en.wikipedia.org/wiki/String_interning) that are
+//! cheap to store and compare (they're secretly [`u32`]s!). This module contains a list of pre-
+//! interned symbol constants that are used in the linter.
+//!
+//! # Symbol Offsets
+//!
+//! The linter allocates symbols in the following layout:
+//!
+//! <table>
+//!     <thead>
+//!         <tr>
+//!             <th>Index</th>
+//!             <th>Note</th>
+//!         </tr>
+//!     </thead>
+//!     <tbody>
+//!         <tr>
+//!             <td>0</td>
+//!             <td rowspan="4">Symbols in <code>rustc_span::sym</code></td>
+//!         </tr>
+//!         <tr>
+//!             <td>1</td>
+//!         </tr>
+//!         <tr>
+//!             <td>2</td>
+//!         </tr>
+//!         <tr>
+//!             <td>...</td>
+//!         </tr>
+//!         <tr>
+//!             <td><code>PREDEFINED_SYMBOLS_COUNT</code></td>
+//!             <td rowspan="2">Symbols in <code>clippy_utils::sym</code></td>
+//!         </tr>
+//!         <tr>
+//!             <td>...</td>
+//!         </tr>
+//!         <tr>
+//!             <td><code>SYMBOL_OFFSET</code></td>
+//!             <td rowspan="2">Symbols in <code>bevy_lint::sym</code></td>
+//!         </tr>
+//!         <tr>
+//!             <td>...</td>
+//!         </tr>
+//!     </tbody>
+//! </table>
```

Comment:
This table is better viewed in Rustdoc's HTML output. Here's a screenshot, though:
![image](https://github.com/user-attachments/assets/d6fc48d1-0b73-479c-9c77-72c99d49b01b)